### PR TITLE
Support IE8 by not using Array.prototype.indexOf() and Object.keys()

### PR DESCRIPTION
**Pull request by bengourley**
I don't know if you're interested in this PR at all but I thought I'd share it since I need IE8 support for my project.

The tests pass and the benchmarks perform well (this module still wins consistently).

Let me know, otherwise happy to maintain a fork.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module.exports = stringify
 stringify.default = stringify
 function stringify (obj) {
-  if (obj !== null && typeof obj === 'object' && typeof obj.toJSON !== 'function') {
+  if (obj !== null && typeof obj === 'object' && (typeof obj.toJSON !== 'function' || obj.toJSON.forceDecirc)) {
```
[the remark this answers is not in the document]
Comment (davidmarkclements) [reply index.js]: Yah without a related bug I'm reluctant to alter this, there's so many side effects that there's a high risk that my reasoning (particularly so long after writing it) will be flawed. In the last minor changes have lead to odd hard to debug issues
in dependants

[the remark this answers is not in the document]
Comment (bengourley) [reply index.js]: Let me tear this out and do it as a separate PR with tests

**Comment by bengourley** [comment]
benchmark.js results

```
Development/fast-safe-stringify - [ie8ify●] » node -v
v8.3.0
Development/fast-safe-stringify - [ie8ify●] » node benchmark.js
inspectBench*10000: 131.636ms
jsonStringifySafeBench*10000: 32.373ms
fastSafeStringifyBench*10000: 22.534ms
inspectCircBench*10000: 174.042ms
jsonStringifyCircSafeBench*10000: 37.511ms
fastSafeStringifyCircBench*10000: 30.706ms
inspectDeepBench*10000: 1622.650ms
jsonStringifySafeDeepBench*10000: 600.763ms
fastSafeStringifyDeepBench*10000: 258.212ms
inspectDeepCircBench*10000: 1650.903ms
jsonStringifySafeDeepCircBench*10000: 616.304ms
fastSafeStringifyDeepCircBench*10000: 262.995ms
jsonStringifyDeepTryFirstBench*10000: 180.370ms
fastSafeStringifyDeepTryFirstBench*10000: 180.833ms
inspectBench*10000: 93.291ms
jsonStringifySafeBench*10000: 25.638ms
fastSafeStringifyBench*10000: 16.298ms
inspectCircBench*10000: 133.067ms
jsonStringifyCircSafeBench*10000: 33.203ms
fastSafeStringifyCircBench*10000: 24.130ms
inspectDeepBench*10000: 1632.501ms
jsonStringifySafeDeepBench*10000: 610.660ms
fastSafeStringifyDeepBench*10000: 248.495ms
inspectDeepCircBench*10000: 1606.765ms
jsonStringifySafeDeepCircBench*10000: 611.899ms
fastSafeStringifyDeepCircBench*10000: 259.084ms
jsonStringifyDeepTryFirstBench*10000: 176.203ms
fastSafeStringifyDeepTryFirstBench*10000: 179.896ms
```

```
Development/fast-safe-stringify - [ie8ify●] » node -v
v6.11.5
Development/fast-safe-stringify - [ie8ify●] » node benchmark.js
inspectBench*10000: 182.870ms
jsonStringifySafeBench*10000: 33.755ms
fastSafeStringifyBench*10000: 21.746ms
inspectCircBench*10000: 251.827ms
jsonStringifyCircSafeBench*10000: 45.011ms
fastSafeStringifyCircBench*10000: 28.819ms
inspectDeepBench*10000: 2327.728ms
jsonStringifySafeDeepBench*10000: 913.607ms
fastSafeStringifyDeepBench*10000: 238.807ms
inspectDeepCircBench*10000: 2373.652ms
jsonStringifySafeDeepCircBench*10000: 925.657ms
fastSafeStringifyDeepCircBench*10000: 244.160ms
jsonStringifyDeepTryFirstBench*10000: 188.388ms
fastSafeStringifyDeepTryFirstBench*10000: 184.350ms
inspectBench*10000: 154.228ms
jsonStringifySafeBench*10000: 24.880ms
fastSafeStringifyBench*10000: 17.974ms
inspectCircBench*10000: 232.048ms
jsonStringifyCircSafeBench*10000: 35.965ms
fastSafeStringifyCircBench*10000: 26.715ms
inspectDeepBench*10000: 2352.140ms
jsonStringifySafeDeepBench*10000: 895.955ms
fastSafeStringifyDeepBench*10000: 220.704ms
inspectDeepCircBench*10000: 2348.619ms
jsonStringifySafeDeepCircBench*10000: 925.994ms
fastSafeStringifyDeepCircBench*10000: 231.228ms
jsonStringifyDeepTryFirstBench*10000: 195.575ms
fastSafeStringifyDeepTryFirstBench*10000: 185.185ms
```

**Comment by davidmarkclements** [comment]
Thanks! For comparison I also need the benchmark results on master from your same machine

**Comment by bengourley** [comment]
Oops, of course!

```
Development/fast-safe-stringify - [62941e8...●] » node -v
v8.3.0
Development/fast-safe-stringify - [62941e8...●] » node benchmark.js
inspectBench*10000: 149.049ms
jsonStringifySafeBench*10000: 34.596ms
fastSafeStringifyBench*10000: 21.321ms
inspectCircBench*10000: 198.177ms
jsonStringifyCircSafeBench*10000: 38.684ms
fastSafeStringifyCircBench*10000: 31.896ms
inspectDeepBench*10000: 1892.228ms
jsonStringifySafeDeepBench*10000: 688.480ms
fastSafeStringifyDeepBench*10000: 262.202ms
inspectDeepCircBench*10000: 1790.321ms
jsonStringifySafeDeepCircBench*10000: 644.733ms
fastSafeStringifyDeepCircBench*10000: 250.429ms
jsonStringifyDeepTryFirstBench*10000: 194.318ms
fastSafeStringifyDeepTryFirstBench*10000: 193.933ms
inspectBench*10000: 107.553ms
jsonStringifySafeBench*10000: 27.819ms
fastSafeStringifyBench*10000: 18.241ms
inspectCircBench*10000: 154.164ms
jsonStringifyCircSafeBench*10000: 35.810ms
fastSafeStringifyCircBench*10000: 26.015ms
inspectDeepBench*10000: 1758.790ms
jsonStringifySafeDeepBench*10000: 633.428ms
fastSafeStringifyDeepBench*10000: 241.303ms
inspectDeepCircBench*10000: 1854.277ms
jsonStringifySafeDeepCircBench*10000: 677.822ms
fastSafeStringifyDeepCircBench*10000: 248.058ms
jsonStringifyDeepTryFirstBench*10000: 188.716ms
fastSafeStringifyDeepTryFirstBench*10000: 189.456ms
```

```
Development/fast-safe-stringify - [62941e8...●] » node -v
v6.11.5
Development/fast-safe-stringify - [62941e8...●] » node benchmark.js
inspectBench*10000: 199.669ms
jsonStringifySafeBench*10000: 37.487ms
fastSafeStringifyBench*10000: 21.458ms
inspectCircBench*10000: 259.714ms
jsonStringifyCircSafeBench*10000: 50.886ms
fastSafeStringifyCircBench*10000: 36.700ms
inspectDeepBench*10000: 2567.015ms
jsonStringifySafeDeepBench*10000: 988.227ms
fastSafeStringifyDeepBench*10000: 271.729ms
inspectDeepCircBench*10000: 2887.870ms
jsonStringifySafeDeepCircBench*10000: 1200.596ms
fastSafeStringifyDeepCircBench*10000: 346.023ms
jsonStringifyDeepTryFirstBench*10000: 212.295ms
fastSafeStringifyDeepTryFirstBench*10000: 215.411ms
inspectBench*10000: 187.292ms
jsonStringifySafeBench*10000: 29.407ms
fastSafeStringifyBench*10000: 45.391ms
inspectCircBench*10000: 309.256ms
jsonStringifyCircSafeBench*10000: 55.155ms
fastSafeStringifyCircBench*10000: 41.717ms
inspectDeepBench*10000: 2840.213ms
jsonStringifySafeDeepBench*10000: 960.570ms
fastSafeStringifyDeepBench*10000: 260.988ms
inspectDeepCircBench*10000: 2632.948ms
jsonStringifySafeDeepCircBench*10000: 1062.526ms
fastSafeStringifyDeepCircBench*10000: 292.512ms
jsonStringifyDeepTryFirstBench*10000: 202.992ms
fastSafeStringifyDeepTryFirstBench*10000: 192.353ms
```

**Comment by davidmarkclements** [comment]
ok so for deep bench we're losing 40ms and deep circ bench 60ms on Node 6

But on Node 8 the perf diff is negligible – since Node 8 has just gone LTS I think this acceptable at this point

Thank you @bengourley

**Comment by davidmarkclements** [comment]
v1.2.1 published

**Pull request merged**